src/controllers/job/Transactions.ts

```diff
@@ -1,12 +1,46 @@
 import { Request, Response } from 'express';
+import * as rp from 'request-promise';
 import * as Models from '../../models';
 
 const Stocks = Models.default.Stocks;
+const url = {
+  FPT:
+    'https://svr2.fireant.vn/api/Data/Markets/HistoricalQuotes?symbol=FPT&startDate=2020-1-20&endDate=2020-4-19'
+};
 
 export default {
   crawl: async (req: Request, res: Response) => {
-    const result = await Stocks.findAll();
+    const stock = await Stocks.findAll();
+
+    stock.forEach(async element => {
```

Comment:
should be write a function 
const crawlByStockCode = (options) => {
};

and in 
stock.forEach(async element => {
   crawlByStockCode({
       stockCode: element.stock_code,
       startDate: req.query.startDate || moment().fomart('YYYY-MM-DD'),
       endDate: req.query.endDate || moment().fomart('YYYY-MM-DD'),
   })
});